### PR TITLE
Bug fix: allow for retrying pcp2json with one pmlogger_interval pause

### PR DESCRIPTION
**Pull request by dustinblack**
## Changes introduced with this PR

Under workflow conditions where the pcp plugin is stopped very quickly before it can collect a `pmlogger_interval` cycle of data, the plugin could fail and lead to a difficult-to-troubleshoot workflow failure. This change removes extraneous commands to explicitly stop the `pmlogger` command, and allows for one retry of the `pcp2json` command after a `pmlogger_interval` sleep.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: arcaflow_plugin_pcp/pcp_plugin.py [hunk 1]
```diff
@@ -144,27 +142,42 @@ def start_pcp(
...
-        try:
-            pcp_out = (
-                (
-                    subprocess.check_output(
-                        pcp2json_cmd,
-                        text=True,
-                        stderr=subprocess.STDOUT,
+        max_retries = 1
+        retries = 0
+        while retries <= max_retries:
```
Comment (jaredoconnell) [inline arcaflow_plugin_pcp/pcp_plugin.py]: Does this still need to be a loop?

Comment (dustinblack) [reply arcaflow_plugin_pcp/pcp_plugin.py]: I implemented a bit of a trick here, but I'm not positive it is the best approach. The `try` code below this does need loop control for at least 2 runs, but within the `while` loop we expect to return the function, either as `success` or as `error`, so the loop should never have a natural exit, which is the reason for the "unknown failure" capture outside of the `while` loop at line 180.

Comment (jaredoconnell) [reply arcaflow_plugin_pcp/pcp_plugin.py]: Okay. I see. Though I do not think it's the most readable.

Comment (dustinblack) [reply arcaflow_plugin_pcp/pcp_plugin.py]: I added code comments to improve this.

**Pull request merged**